src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    public static final int MAX_FIGURE = 3;
+    public static final int MAX_SIDE = 20;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(MAX_FIGURE);
+        int firstNumber = random.nextInt(MAX_SIDE);
+        int secondNumber = random.nextInt(MAX_SIDE);
+        int thirdNumber = random.nextInt(MAX_SIDE);
```

Comment:
Don't create those variables here, because you don't need all of them. (e.g. for Circle you need only firstNumber).

Instead, move it's creation to the corresponding `switch` blocks, and also move the creation of the figure to the separate methods.

Also, let's change the naming. For example, use `width`, `height` instead of `firstNumber` etc.